sim/midas/src/main/scala/midas/core/FPGATop.scala

```diff
@@ -337,12 +414,18 @@ class FPGATopImp(outer: FPGATop)(implicit p: Parameters) extends LazyModuleImp(o
     "MEM_LEN_BITS"   -> AXI4Parameters.lenBits,
     "MEM_RESP_BITS"  -> AXI4Parameters.respBits,
     // Address width of the aggregated host-DRAM space
-    "DMA_ID_BITS"    -> dma.params.idBits,
-    "DMA_ADDR_BITS"  -> dma.params.addrBits,
-    "DMA_DATA_BITS"  -> dma.params.dataBits,
-    "DMA_STRB_BITS"  -> dma.params.dataBits / 8,
-    "DMA_BEAT_BYTES" -> p(DMANastiKey).dataBits / 8,
-    "DMA_SIZE"       -> log2Ceil(p(DMANastiKey).dataBits / 8),
-  ) ++ Seq.tabulate[(String, Long)](p(HostMemNumChannels))(idx => s"MEM_HAS_CHANNEL${idx}" -> 1)
+    "DMA_ID_BITS"    -> dma.map(_.params.idBits)      .getOrElse(0).toLong,
+    "DMA_ADDR_BITS"  -> dma.map(_.params.addrBits)    .getOrElse(0).toLong,
+    "DMA_DATA_BITS"  -> dma.map(_.params.dataBits)    .getOrElse(0).toLong,
+    "DMA_STRB_BITS"  -> dma.map(_.params.dataBits / 8).getOrElse(0).toLong,
```

Comment:
Should this be `DMA_STRB_BYTES`?

Reply:
Confusing yes, but it's the bitwidth of the strobe field, which masks equally many bytes.